src/ring_theory/eisenstein_criterion.lean

```diff
@@ -0,0 +1,101 @@
+/-
+Copyright (c) 2020 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+-/
+import ring_theory.ideal_operations
+import data.polynomial
+import tactic.apply_fun
+import ring_theory.prime
+/-!
+# Eisenstein's criterion
+
+A proof of a slight generalisation of Eisenstein's criterion for the irreducibility of
+a polynomial over an integral domain.
+-/
+
+open polynomial ideal.quotient
+variables {R : Type*} [integral_domain R]
+
+/-- If `f` is a non constant polynomial with coefficients in `R`, and `P` is a prime ideal in `R`,
+then if every coefficient in `R` except the leading coefficient is in `P`, and
+the trailing coefficient is not in `P^2` and no non units in `R` divide `f`, then `f` is
+irreducible. -/
+theorem eisenstein_criterion {f : polynomial R} {P : ideal R} (hP : P.is_prime)
```

Comment:
Should we rename this to `irreducible_of_eisenstein_criterion`? I noticed that in `quadratic_reciprocity` there are also several lemmas that have Namey names instead of following the mathlib naming convention. In particular, we also have an `eisenstein_lemma` already.